src/Client.php

```diff
@@ -90,16 +89,16 @@ public function captureOrderTransaction(string $id): bool
     {
         $order = $this->getOrder(id: $id);
 
-        if ($order->getStatus() !== 'completed') {
+        if ($order->getStatus()->get() !== 'completed') {
             throw new InvalidOrderStatusException(
-                actual: $order->getStatus(), expected: 'completed'
+                actual: $order->getStatus()->get(), expected: 'completed'
```

Comment:
I find `$order->getStatus()->get()` a bit alogical, I suggest to use:
1) `$order->status()->get()`
2) `$order->getStatus()`

Reply:
> $order->status()->get()

I'm not sure that access to properties should be made public. All user needs are covered by one or another method of working with data.

Public access can harm the user himself.